pkg/patch/patch.go

```diff
@@ -97,3 +110,35 @@ func GeneratePatch(meta Meta) ([]byte, error) {
 
 	return []byte(fmt.Sprintf("[%s]", strings.Join(patches, ","))), nil
 }
+
+func appMeshCNIAnnotationsPatch(meta Meta) []string {
+	newAnnotations := map[string]string{
+		"appmesh.k8s.aws/egressIgnoredIPs":   meta.Init.IgnoredIPs,
+		"appmesh.k8s.aws/egressIgnoredPorts": meta.Init.EgressIgnoredPorts,
+		"appmesh.k8s.aws/ports":              meta.Init.Ports,
+	}
+	return annotationsPatches(meta.PodAnnotations, newAnnotations)
+}
+
+func annotationsPatches(existingAnnotations map[string]string, newAnnotations map[string]string) (patches []string) {
+	for key, value := range newAnnotations {
+		if existingAnnotations == nil {
+			//first one will be create, subsequent will be updates
+			existingAnnotations = map[string]string{}
```

Comment:
I think you might simplify this function by just initializing the existingAnnotations map outside of the top-level `for` loop. Something like this might work:

```go
if existingAnnotations == nil {
    existingAnnotations = map[string]string{}
}
var op string
for key, value := range newAnnotations {
    if _, found := existingAnnotations[key]; !found {
        op = "add"
    } else {
        op = "replace"
    }
    patches = append(patches, fmt.Sprintf(updateAnnotation, op, escapeJSONPointer(key), value))
}
return patches
```

you can then actually get rid of the `createAnnotation` const.

Reply:
I tried that but there is a gotcha in how json-patch works (@stefanprodan correct me here). 
If you look closely, `createAnnotation` and `updateAnnotation w add` are different w.r.t `path` field. This is to handle the case where there are no annotations on pod-spec and we need to add `annotations` field. That said, annotations will never be nil in this context, so the `== nil` code is unreachable. I can remove it and add a doc if that makes it readable.